### PR TITLE
Ac 94 create order route

### LINKED ISSUE #94: feat(f6): Create order route
- Create an order route
- Update apiRouter to use new order route
- Configure new order route(get, delete, update, etc.)

### PR DESCRIPTION
**Pull request by sirpuffin1**
### Changes
1. create an order route
2. add order route to API router
3. add CRUD operations to order route

### Purpose
Allow users to create orders

### Approach
Very carefully

### Learning
Teaching moment

Closes #94

### CODE REVIEW

File: src/server/routers/order.route.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    res.send(orderList)
+})
+
+orderRouter.post("/create", async function(req,res) {
+    const {orderItems, shippingAddress, user} = req.body;
+    const items = await OrderItemModel.insertMany(orderItems)
+
+    const totalPrices = await Promise.all(items.map(async (orderItemId)=> {
+        const orderItem = await OrderItemModel.findById(orderItemId).populate('product', 'price')
+        const totalPrice = orderItem.product.price * orderItem.quantity
```
Comment (brycejjacks) [inline src/server/routers/order.route.ts:21]: wouldnt the price already be on the product so you only need to populate the product ... i could very well be wrong i am asking?

File: src/server/routers/order.route.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import express from "express";
+import { OrderItemModel } from "../schemas/order-items.schema.js";
+import { OrderModel } from "../schemas/order.schema.js";
+export const orderRouter = express.Router();
+
+orderRouter.get('/', async(req, res) => {
+    const orderList = await OrderModel.find()
+
+    if(!orderList) {
```
Comment (brycejjacks) [inline src/server/routers/order.route.ts]: i think oderList is always truthy because an empty array evaluates to true.. so I don't know how much function this has .. and if it did return false .. wouldnt it be responding twice? so you might get that "cannot set headers after sent" error or what ever ??

File: src/server/routers/order.route.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+})
+
+orderRouter.get('/get/userorders/:userid', async (req, res) => {
+    const userOrderList = await OrderModel.find({user: req.params.userid}).populate({
+        path: 'orderItems', populate: {
+            path: 'product'
+        }
+    })
+
+    if(!userOrderList) {
```
Comment (brycejjacks) [inline src/server/routers/order.route.ts]: Here is another example of how an empty array is truthy... this status 500 never sends... because even if the user has no orders it return [] try something like this....
```suggestion
    if(!userOrderList._id) {
```

**Review by brycejjacks — APPROVED** [review]
Ok i checkout out this branch and tried some of the routes with postman ... try to have your viewers in mind when you make a PR tho, because this PR is a pain in the ass to test lol ... im not finna type all that shipping info and orderitems stuff in json into postman lol.. also some of the functions are hard to follow for a measly apprentice like myself, with the arrow functions awaiting and mapping and as a pipe with a promise lol sheesh ... but i do see the creativity in how you have to go into orderitems when someone touches orders. Good job overall be very very cautions of those comments about arrays being truthy .. you did that a few times ... so just go double check to make sure the 'if (!foo)'s are valid

File: src/server/routers/order.route.ts (hunk 1 above)
Comment (sirpuffin1) [reply src/server/routers/order.route.ts:21]: You're absolutely right just fixed that thanks. Also yeah that's my bad lol if anything next time I'll comment in a block of code that you can just copy and paste into postman

File: src/server/routers/order.route.ts (hunk 2 above)
Comment (sirpuffin1) [reply src/server/routers/order.route.ts]: Yeah you're right I completely forgot about truthy values but I updated the code to just test the length of orderLIst and yeah should be fine now. thanks

File: src/server/routers/order.route.ts (hunk 3 above)
Comment (sirpuffin1) [reply src/server/routers/order.route.ts]: ._id didn't want to work since buuuuut if(userOrderList.length === 0) did work. thanks again xoxoxo

**Review by brycejjacks — APPROVED** [review]
nice changes i reviewed again and im merging
